syntax = "proto3";
package Protocol;

import "Struct.proto";

message Ping
{
	string msg = 1;
}

message C_Enter
{
	string enter_key = 1;
}

message S_Enter
{
	bool result = 1;
	uint32 enter_id = 2;
}

message C_Exit
{
	uint32 enter_id = 1;
}

message S_Exit
{
	bool result = 1;
	uint32 enter_id = 2;
}

message C_Spawn
{
	uint32 spawn_count = 1;
}

message S_Spawn
{
	bool result = 1;
	repeated PlayerInfo player_infos = 2;
}

message S_Despawn
{
	repeated uint32 player_ids = 2;
}